src/Translate/Helpers/ComponentConfigurationTranslationHelper.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace OpenDialogAi\Core\Translate\Helpers;
+
+use Illuminate\Support\Str;
+use OpenDialogAi\Core\Components\Configuration\ComponentConfiguration;
+use OpenDialogAi\Core\ComponentSetting;
+use OpenDialogAi\Core\Translate\Facades\TranslateService;
+
+class ComponentConfigurationTranslationHelper
+{
+    /**
+     * Adds or updates translatable strings into the translate table, for this ComponentConfiguration
+     *
+     * @param ComponentConfiguration $componentConfiguration
+     * @param string $language
+     */
+    public static function addOrUpdateComponentConfigurationTranslations(
+        ComponentConfiguration $componentConfiguration,
+        string $language = 'en'
+    ): void {
+        $configurations = $componentConfiguration->configuration;
+
+        foreach ($configurations as $configurationName => $configurationSetting) {
+            if (!is_array($configurationSetting)) {
+                continue;
+            }
+
+            foreach ($configurationSetting as $key => $value) {
+                /** @var ComponentSetting $setting */
+                $setting = ComponentSetting::where('name', $key)->first();
+                if ($setting?->translatable) {
+                    $translateId = TranslateService::addOrUpdateConfigTranslation(
+                        $setting->name,
+                        $componentConfiguration->scenario_id,
+                        $value,
+                        $language
+                    );
+                    $configurations[$configurationName][$key] = $translateId;
+                }
+            }
+        }
+        $componentConfiguration->configuration = $configurations;
```

Comment:
```suggestion
        $componentConfiguration->configuration = $configurations;
        $componentConfiguration->save();
```

We're currently not persisting the config changes, `save()` will make this happen

Reply:
No, this should not be the responsibility of the helper. Helper just pulls the translatable strings, adds them into translation table and updates the translatable string with ids on the ComponentConfiguration. After that, the calling code can decide wether to save the configuration or not.

Reply:
For this to work the $componentConfiguration needs to be passed by reference, or returned

Reply:
Hey @oliversarfas, that works, because in php objects are passed by reference by default. [Here's](https://onlinephp.io/c/75775) a quick example.